Kernel/Process.cpp

```diff
@@ -2282,6 +2282,47 @@ int Process::sys$getpeername(int sockfd, sockaddr* addr, socklen_t* addrlen)
     return 0;
 }
 
+int Process::sys$sched_setparam(pid_t pid, struct sched_param* param)
+{
+    if (!validate_read_typed(param))
+        return -EFAULT;
+
+    auto* peer = this;;
+    if (pid != 0)
+        peer = Process::from_pid(pid);
```

Comment:
So, because the Process list can be mutated by the scheduler, we actually can't take a `Process*` like this and expect it to stay valid for the rest of this function.
Since we need `peer` until the very end of this function, we should just slap an `InterruptDisabler` RAII object before the call to `Process::from_pid()` (and make sure it stays alive until the function exits.)
That way the code is guaranteed to execute serially.